scripts/pi-hole/php/teleporter.php

```diff
@@ -300,171 +300,137 @@ function archive_add_directory($path, $subdir = '')
-}
-
-function noun($num)
+function load_teleport_archive($filename, $options)
 {
-    if ($num === 1) {
-        return ' entry';
-    }
-
-    return ' entries';
-}
-
-if (isset($_POST['action'])) {
-    if ($_FILES['zip_file']['name'] && $_POST['action'] == 'in') {
-        $filename = $_FILES['zip_file']['name'];
-        $source = $_FILES['zip_file']['tmp_name'];
-        $type = mime_content_type($source);
-
-        // verify the file mime type
-        $accepted_types = array('application/gzip', 'application/tar', 'application/x-compressed', 'application/x-gzip');
-        $mime_valid = in_array($type, $accepted_types);
-
-        // verify the file extension (Looking for ".tar.gz" at the end of the file name)
-        $ext = array_slice(explode('.', $filename), -2, 2);
-        $ext_valid = strtolower($ext[0]) == 'tar' && strtolower($ext[1]) == 'gz' ? true : false;
-
-        if (!$ext_valid || !$mime_valid) {
-            exit('The file you are trying to upload is not a .tar.gz file (filename: '.htmlentities($filename).', type: '.htmlentities($type).'). Please try again.');
-        }
-
-        $fullfilename = sys_get_temp_dir().'/'.$filename;
-
-        if (!move_uploaded_file($source, $fullfilename)) {
-            exit('Failed moving '.htmlentities($source).' to '.htmlentities($fullfilename));
-        }
-
-        $archive = new PharData($fullfilename);
-
-        $importedsomething = false;
-        $fullpiholerestart = false;
-        $reloadsettingspage = false;
-
-        $flushtables = isset($_POST['flushtables']);
+    $importedsomething = false;
+    $fullpiholerestart = false;
+    $reloadsettings = false;
+
+    $flushtables = isset($options['flushtables']);
+    $blacklist = isset($options['blacklist']);
+    $whitelist = isset($options['whitelist']);
+    $regex_blacklist = isset($options['regex_blacklist']);
+    $regex_whitelist = isset($options['regex_whitelist']);
+    $regexlist = isset($options['regexlist']);
+    $auditlog = isset($options['auditlog']);
+    $adlist = isset($options['adlist']);
+    $group = isset($options['group']);
+    $client = isset($options['client']);
+    $staticdhcpleases = isset($options['staticdhcpleases']);
+    $localdnsrecords = isset($options['localdnsrecords']);
+    $localcnamerecords = isset($options['localcnamerecords']);
+
+    $endline = php_sapi_name() === 'cli' ? "\n" : "<br>\n";
+
+    try {
+        $archive = new PharData($filename);
 
         foreach (new RecursiveIteratorIterator($archive) as $file) {
-            if (isset($_POST['blacklist']) && $file->getFilename() === 'blacklist.txt') {
+            if ($blacklist && $file->getFilename() === 'blacklist.txt') {
                 $num = archive_insert_into_table($file, 'blacklist', $flushtables);
-                echo 'Processed blacklist (exact) ('.$num.noun($num).")<br>\n";
+                echo 'Processed blacklist (exact) ('.$num.noun($num).")$endline";  // it is possible to remove the <br>, but is it worth it?
```

Comment:
I think using `$endline` like you did it's fine.

Please, remove the comment.

```suggestion
                echo 'Processed blacklist (exact) ('.$num.noun($num).")$endline";
```